### PR TITLE
[tflite] python script for object detection w/ SSD

### PR DESCRIPTION
**Pull request by freedomtan**
Add a TFLite Python script that uses SSD to detect objets.

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @jch1: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @jch1: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 29 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @jch1: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @jch1: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 29 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @jch1: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 44 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @jch1: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 59 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @jch1: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 74 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @jch1: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 89 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @jch1: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 104 days with no activity and the `awaiting review` label has been applied.

**Comment by gbaned** [comment]
@jch1, @achowdhery Hi, Could you PTAL and approve

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @jch1, @achowdhery: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/lite/examples/python/ssd_object_detection.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  file_name = "/tmp/image2.jpg"
+  model_file = "/tmp/detect.tflite"
+  label_file = "/tmp/labelmap.txt"
+  input_mean = 127.5
+  input_std = 127.5
+  floating_model = False
+  show_image = False
+
+  parser = argparse.ArgumentParser()
+  parser.add_argument("--image", help="image to be classified")
```
Comment (Dayananda-V) [inline tensorflow/lite/examples/python/ssd_object_detection.py]: suggest to define default value under argument declaration and parse argument with short key as well.
same comment can be applicable for all argument.

File: tensorflow/lite/examples/python/ssd_object_detection.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  interpreter.allocate_tensors()
+
+  input_details = interpreter.get_input_details()
+  output_details = interpreter.get_output_details()
+  if args.verbose:
+    print(input_details)
+    print(output_details)
+
+  # check the type of the input tensor
+  if input_details[0]['dtype'] == type(np.float32(1.0)):
```
Comment (Dayananda-V) [inline tensorflow/lite/examples/python/ssd_object_detection.py]: hope this is can be replace with better one as 
`input_details[0]['dtype'] == np.float32`

File: tensorflow/lite/examples/python/ssd_object_detection.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  detected_scores = interpreter.get_tensor(output_details[2]['index'])
+  num_boxes = interpreter.get_tensor(output_details[3]['index'])
+
+  if args.verbose:
+    print("num_boxes:", num_boxes[0])
+    print("detected boxes:", detected_boxes)
+    print("detected classes:", detected_classes)
+    print("detected scores:", detected_scores)
+
+  if show_image:
```
Comment (Dayananda-V) [inline tensorflow/lite/examples/python/ssd_object_detection.py]: keep 'matplotlib.pyplot' package under show_image flag.

File: tensorflow/lite/examples/python/ssd_object_detection.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+      ax.add_patch(rect)
+      label_string = labels[int(detected_classes[0][r])+1]
+      score_string = '{0:2.0f}%'.format(detected_scores[0][r] * 100)
+      ax.text(left, top, label_string + ': ' + score_string, \
+              fontsize=6, bbox=dict(facecolor='y', edgecolor='y', alpha=0.5))
+
+  if show_image:
+    ax.imshow(img)
+    plt.title(model_file)
+    plt.show()
```
Comment (Dayananda-V) [inline tensorflow/lite/examples/python/ssd_object_detection.py]: Suggest to provide save test result when show_image is disable as default option. Same will update on document as well.

**Review by Dayananda-V — CHANGES_REQUESTED** [review]

**Comment by jdduke** [comment]
Hi @freedomtan, thanks for the PR. We're no longer adding additional examples in lite/examples; most of our examples have moved to the new [TF examples repo](https://github.com/tensorflow/examples/tree/master/lite). We're still trying to determine how we want to present Python-based examples. Once we have that figured out, I'll reopen the PR and we can proceed. Thanks again for your contribution, hopefully we can make this work.

**Comment by freedomtan** [comment]
@jdduke Thanks. As you know, Python examples won't work on Android devices, but they work on many other environments and could helpful when testing if models work well with TFLite. I saw people asking for example scripts for object detection and segmentation several times before.

**Comment by jdduke** [comment]
Absolutely, Python examples are important and we'd like to expand the set of such examples. Once we have a path forward for how/where we'd like to host them we can get back to you about this particular contribution. Thanks again for your patience.

**Pull request closed without merging**